app/controllers/instructeurs/dossiers_controller.rb

```diff
@@ -331,6 +331,31 @@ def restore
       end
     end
 
+    def reaffectation
+      @dossier = current_instructeur.dossiers.find(params[:dossier_id])
+
+      @groupe_instructeur = @dossier.groupe_instructeur
+
+      @groupes_instructeurs = Kaminari.paginate_array(@groupe_instructeur.other_groupe_instructeurs)
+        .page(params[:page])
+        .per(ITEMS_PER_PAGE)
+    end
+
+    def reaffecter
+      dossier = current_instructeur.dossiers.find(params[:dossier_id])
+
+      procedure = current_instructeur.procedures.find(params[:procedure_id])
+
+      new_group = procedure.groupe_instructeurs.find_by(id: params[:groupe_instructeur_id])
+
```

Comment:
```suggestion
      dossier = current_instructeur.dossiers.find(params[:dossier_id])
      
      new_group = dossier
        .procedure
        .groupe_instructeurs.find(params[:groupe_instructeur_id])
```

je retire l'intermédiaire `procedure = ` pour être sur qu'on travaille sur la procedure du dossier, et j'utilise un `find` au lieu du `find_by` vue qu'on cherche par id et qu'on veut que ca pète si on le trouve pas.